### PR TITLE
Feature gate the `Scan` trait and integrate buildbot support

### PR DESCRIPTION
**Pull request by jacob-hughes**
Buildbot should run `rustc` test suite

### CODE REVIEW

**Comment by jacob-hughes** [comment]
bors ping

**Comment by jacob-hughes** [comment]
bors ping

File: .buildbot.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+#
+# Build script for continuous integration.
+
+./x.py clean  # We don't clone afresh to save time and bandwidth.
```
Comment (vext01) [inline .buildbot.sh]: Ah, I think we do clone afresh now. So you probably don't need any cleaning steps. Although it doesn't hurt to have them there.

We had issues with trying to re-use the repo.

If you want to verify that yourself, the relevant line in the buildbot config is (I think):
```
buildbot_sh_factory.addStep(steps.Git(repourl=get_repourl, mode='full', method='
clobber', shallow=True))
```

Comment (vext01) [reply .buildbot.sh]: And yeah, I know you copied this from ykrustc. I should make the changes there too!

Comment (jacob-hughes) [reply .buildbot.sh]: Sure, I've removed the `./x.py clean` now.

Comment (jacob-hughes) [reply .buildbot.sh]: Sure, I've removed the `./x.py clean` now.

File: .buildbot.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+#
+# Build script for continuous integration.
+
+git clean -dffx # If upstream removes a submodule, remove the files from disk.
```
Comment (vext01) [inline .buildbot.sh]: This can also go, I think.

Comment (jacob-hughes) [reply .buildbot.sh]: Gone

**Comment by vext01** [comment]
Ready to squash then?

**Comment by jacob-hughes** [comment]
Sure, I'll squash now

**Comment by jacob-hughes** [comment]
Squashed!

**Comment by vext01** [comment]
It's building!

**Comment by jacob-hughes** [comment]
Pushed some fixes to hopefully get buildbot to stop complaining

**Comment by jacob-hughes** [comment]
bors try

**Comment by jacob-hughes** [comment]
bors try

**Comment by jacob-hughes** [comment]
I've made some fixes to get this building, but will need to squash and add a description of the fixes to the fix commits

**Comment by vext01** [comment]
If you are all pushed up, please squash.

**Comment by jacob-hughes** [comment]
Squashed

**Comment by vext01** [comment]
I should have borsed this. My bad.

Here's a build against the last commit jake pushed:
http://bencher12.soft-dev.org:8010/#/builders/2/builds/131

**Comment by vext01** [comment]
Let's try that again, shall we?
http://bencher12.soft-dev.org:8010/#/builders/2/builds/132

**Comment by ltratt** [comment]
It worked, so all is good!

**Pull request merged**